packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -44,6 +51,42 @@ const defaultProps = {
   buttonAlign: 'start',
   isOpen: false,
   preview: undefined,
+  focusRef: null,
+};
+
+/**
+ * Disables focus styles for normally unfocusable elements.
+ */
+const disableFocusStyles = (event) => {
+  event.currentTarget.setAttribute('data-focus-styles-enabled', 'false');
+};
+
+/**
+ * Remove attributes that makes element focusable.
+ */
+const removeAttributes = (event) => {
+  event.currentTarget.removeAttribute('tabindex');
+  event.currentTarget.removeAttribute('role');
+};
+
+/**
+ * Focus element.
+ */
+const focusElement = (element) => {
+  element.setAttribute('data-focus-styles-enabled', 'true');
+  element.addEventListener('blur', disableFocusStyles);
+  // try to focus the element.
+  element.focus();
+  // if element is not focusable, add atributes that make it focusable, then focus again.
+  if (document.activeElement !== element) {
+    element.setAttribute('tabIndex', '-1');
+    element.setAttribute('role', 'group');
+    element.addEventListener('blur', (event) => {
+      disableFocusStyles(event);
+      removeAttributes(event);
+    });
+    element.focus();
+  }
```

Comment:
Is this the correct approach?  I would expect the consumer to do the work necessary to ensure that the element is focusable.  I don't believe the platform component should be making updates to the DOM of our consumers.

Reply:
This case is unique as we need to make a normally not intractable DOM element like <span> placed inside the <p> to get focus for a short period of time (once the full content was revealed) to be announced by an assistive technologies. Then, once user moves away from that DOM element, it should lose that focus, as it should not be able to get it back. The logic that has been implemented is following:
1. Try to set focus to the DOM element of interest.
2. Check if element got focus (meaning the DOM element is focus-able as is, like link). If yes, we do not touch the DOM element at all, the code block in question is omitted as `if (document.activeElement !== element)` statement returns `false`.
3. If no, we temporarily set the `tabindex` and `role` to the DOM element, but we also attach the event listener, which would remove them on blur event. After DOM element loses focus it gets back to the exact state it was.

Summarizing that, focus-able DOM elements stay intact in all cases, not focus-able DOM elements' attributes change their tab-index for a short period of time, then they got back to their initial state.

Reply:
If the element is not focusable, then you cannot set focus to it and would not get the style.  We should not have to add an event listener to the DOM directly, right?

Reply:
I might have misunderstood the question, but I got it as why we need the event listener here. Let me clarify "If the element is not focusable" part here. Let's assume the element is `<span>`. Initially it's not focusable. 
Once the content is revealed, it has to become focusable to get focus so that assistive technologies read it out, that is why `tabindex= -1` attribute is added. Now it's `<span tabindex="-1">`, and it is focusable. 
While `tabindex=-1` does not allow the element to get focus by tabbing, it still can get focus by clicking on the element with the mouse. It is a `<span>` element that does nothing on click, it should not get focus more than once upon opening the show-hide content. Hence, span should get back to its usual non-focusable self. 
Per that, at the moment when the 'tabindex=-1' is being added to the element, the event listener onBlur event is added as well. That even listener removes 'tabindex=-1' returning span to its initial state.